application/models/Traits/PermissionTrait.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+trait PermissionTrait
+{
+
+    /**
+     * Get the owner id of this record
+     * Used for Permission, to be extendable for each model with owner
+     * @return integer|null
+     */
+    public function getOwnerId()
+    {
+        return null;
+    }
+
+    /**
+     * Get Permission data for Permission object
+     * @param string $key
+     * @return array
+     */
+    public static function getPermissionData($key = null)
+    {
+        return array();
+    }
+
+    /**
+     * Get minimal permission name (for read value)
+     * @return null|string
+     */
+    public static function getMinimalPermissionRead()
+    {
+        return null;
```

Comment:
Maybe use:

```php
throw new Exception('Not implemented');
```

Reply:
No

SureysInGroup didn't have minimal Permission : it's desired behaviour. 

https://github.com/LimeSurvey/LimeSurvey/blob/4b569fc2e6d55733176f782695dd84fbf4fd8206/application/models/Permission.php#L344